ddtrace/contrib/molten/patch.py

```diff
@@ -0,0 +1,143 @@
+    """
+    if getattr(molten, '_datadog_patch', False):
+        return
+    setattr(molten, '_datadog_patch', True)
+
+    pin = Pin(
+        service=config.molten['service_name'],
+        app=config.molten['app'],
+        app_type=config.molten['app_type'],
+    )
+
+    # add pin to module since many classes use __slots__
+    pin.onto(molten)
+
+    _w = wrapt.wrap_function_wrapper
+    _w('molten', 'App.__init__', patch_app_init)
+    _w('molten', 'App.__call__', patch_app_call)
+    _w('molten', 'Router.add_route', patch_add_route)
+
+def trace_func(resource):
+    @wrapt.function_wrapper
+    def _trace_func(wrapped, instance, args, kwargs):
+        pin = Pin.get_from(molten)
+
+        if not pin or not pin.enabled():
+            return wrapped(*args, **kwargs)
+
+        with pin.tracer.trace(func_name(wrapped), service=pin.service, resource=resource):
+            return wrapped(*args, **kwargs)
+
+    return _trace_func
+
+def trace_middleware(middleware):
+    @wrapt.function_wrapper
+    def _trace_middleware(wrapped, instance, args, kwargs):
+        pin = Pin.get_from(molten)
+        if not pin or not pin.enabled():
+            return wrapped(*args, **kwargs)
+
+        name = None
+        if inspect.isfunction(wrapped):
+            name = wrapped.__name__
+        else:
+            name = type(wrapped).__name__
+
+        with pin.tracer.trace(name, service=pin.service):
+            return trace_func(name)(wrapped(*args, **kwargs))
+
+    return _trace_middleware(middleware)
+
+def patch_start_response(start_response):
+    @wrapt.function_wrapper
+    def _start_response(wrapped, instance, args, kwargs):
+        pin = Pin.get_from(molten)
+        span = pin.tracer.current_root_span()
+        status, headers, exc_info = args
+        status_code, _, _ = status.partition(' ')
+        span.set_tag(http.STATUS_CODE, status_code)
+        return wrapped(*args, **kwargs)
+
+    return _start_response(start_response)
+
+def patch_add_route(wrapped, instance, args, kwargs):
+    def _wrap(route_like, prefix="", namespace=None):
+        # avoid patching non-Route, e.g. Include
+        if not isinstance(route_like, molten.Route):
+            return wrapped(*args, **kwargs)
+
+        resource = '{} {}'.format(
+            route_like.method,
+            route_like.name or route_like.template
+        )
+
+        # patch handler for route
+        route_like.handler = trace_func(resource)(route_like.handler)
+
+        return wrapped(route_like, prefix=prefix, namespace=namespace)
+
+    return _wrap(*args, **kwargs)
+
+def patch_app_call(wrapped, instance, args, kwargs):
+    pin = Pin.get_from(molten)
+    if not pin or not pin.enabled():
+        return wrapped(*args, **kwargs)
+
+    params, start_response = args
+    start_response = patch_start_response(start_response)
+    method = params.get('REQUEST_METHOD')
+    path = params.get('PATH_INFO')
+    resource = u'{} {}'.format(method, path)
```

Comment:
We will want this to be `{method} {route.template}`.

The reason is because we want to 1) reduce the cardinality in resource names and 2) allow grouping of all requests for the same handler.

e.g. We want the resource to be like `GET /users/{user_id}/products/{product_id}` instead of `GET /users/10/products/10`

We can/should make the full path available as part of the `http.URL` tag (like you have below).